src/shared/context/theme-provider/ThemeProvider.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import themeStyles from '../../styles';
+import PropTypes from 'prop-types';
+
+export const ThemeContext = React.createContext();
+
+const ThemeContextProvider = ({ children }) => (
+    <ThemeContext.Provider value={ { themeStyles } }>
+        { children }
+    </ThemeContext.Provider>
+);
+
+ThemeContextProvider.propTypes = {
+    children: PropTypes.node.isRequired,
```

Comment:
In general, `children` will always be a React element, no?

```suggestion
    children: PropTypes.element.isRequired,
```